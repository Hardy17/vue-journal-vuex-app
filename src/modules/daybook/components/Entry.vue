<template>
  <div class="entry-container mb-3 pointer p-2"
  @click="$router.push({name:'entry',params:{id:10}})"
  >

      <!--Titulo -->
    <div class="entry-title d-flex">
        <span class="text-success fs-5 fw-bold">25</span>
        <span class="mx-1 fs-5">Enero</span>
        <span class="mx-2 fw-light">2022, Martes</span>
    </div>
    <div class="entry-description">
       Lorem ipsum dolor sit, amet consectetur adipisicing elit. Quod excepturi delectus ab aliquid repellendus itaque, doloremque vero. Rem asperiores magnam fugit? Commodi aliquam aliquid mollitia blanditiis perferendis numquam dignissimos consectetur. 
    </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container{
  border-bottom:1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover{
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2 all ease-in;
  }
  .entry-description{
    font-size: 12px;
  }
}
</style>