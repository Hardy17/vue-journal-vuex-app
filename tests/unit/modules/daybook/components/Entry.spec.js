import { shallowMount } from "@vue/test-utils"
import Entry from '@/modules/daybook/components/Entry'
import { journalState } from "../mocks/test-journal-state"
describe('Pruebas en el entry componet',()=>{
    
    const mockRouter={
        push:jest.fn()
    }
    const wrapper=shallowMount(Entry,{
        props:{
            entry:journalState.entries[0]
        },
        global:{
            mocks:{
                $router:mockRouter
            }
        }
    })
  
    test('Debe Hacer el Match con el snapshot',()=>{
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('Debe redireccionar al hacer el click en el entry-container',()=>{
       const entryContainer= wrapper.find('.entry-container')
       entryContainer.trigger('click')
       expect(mockRouter.push).toHaveBeenCalledWith({
           name:'entry',
           params:{
               id:journalState.entries[0].id
           }
       })
    })
    test('pruebas en las propiedades computadas',()=>{
        //wrapper.vm ------>ver las propiedas computadas
        //day:23
        //month:Julio
        //yearDay:'2021,Viernes'
        expect(wrapper.vm.day).toBe(22)
        expect(wrapper.vm.month).toBe('Febrero')
        expect(wrapper.vm.yearDate).toBe('2022, Martes')
    })
})