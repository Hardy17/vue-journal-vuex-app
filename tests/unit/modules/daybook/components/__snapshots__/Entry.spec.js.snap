// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry componet Debe Hacer el Match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!--Titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">22</span><span class="mx-1 fs-5">Febrero</span><span class="mx-2 fw-light">2022, Martes</span></div>
  <div class="entry-description">Hola el proximo paso es aprende Nuxt para vue y serve side redering</div>
</div>
`;
